backends/ipvsfullstate/common.go

```diff
@@ -1,38 +1,73 @@
 package ipvsfullsate
 
 import (
+	v1 "k8s.io/api/core/v1"
+	netutils "k8s.io/utils/net"
 	"sigs.k8s.io/kpng/api/localv1"
 	"sigs.k8s.io/kpng/backends/ipvsfullstate/internal/ipvs"
 )
 
-// getClusterIPs safely returns ClusterIPs associated with the service.
-func getClusterIPs(service *localv1.Service) []string {
+// getClusterIPs returns ClusterIPs for given IPFamily associated with the service.
+func getClusterIPs(service *localv1.Service, ipFamily v1.IPFamily) []string {
 	IPs := make([]string, 0)
 	if service.IPs.ClusterIPs != nil {
-		return service.IPs.ClusterIPs.V4
+		if ipFamily == v1.IPv4Protocol {
+			return service.IPs.ClusterIPs.GetV4()
+		} else if ipFamily == v1.IPv6Protocol {
+			return service.IPs.ClusterIPs.GetV6()
+		}
 	}
 	return IPs
 }
 
-// getExternalIPs safely returns ExternalIPs associated with the service.
-func getExternalIPs(service *localv1.Service) []string {
+// getExternalIPs safely returns ExternalIPs for given IPFamily associated with the service.
+func getExternalIPs(service *localv1.Service, ipFamily v1.IPFamily) []string {
 	IPs := make([]string, 0)
 	if service.IPs.ExternalIPs != nil {
-		return service.IPs.ExternalIPs.V4
+		if ipFamily == v1.IPv4Protocol {
+			return service.IPs.ExternalIPs.GetV4()
+		} else if ipFamily == v1.IPv6Protocol {
+			return service.IPs.ExternalIPs.GetV6()
+		}
 	}
 	return IPs
 }
 
-// getNodeIPs safely returns all Node IPs.
-func getNodeIPs() []string {
-	return *NodeAddresses
+// getNodeIPs safely returns all Node IPs for given IPFamily.
+func getNodeIPs(ipFamily v1.IPFamily) []string {
+	IPs := make([]string, 0)
+	for _, ip := range *NodeAddresses {
+		if ipFamily == v1.IPv4Protocol && netutils.IsIPv4String(ip) {
```

Comment:
wait cant this just be if IPv4 or IPv6 then
```
IPs = append(...
```

Reply:
```
func getExternalIPs(service *localv1.Service, ipFamily v1.IPFamily) ([]string, error) {
	if service.IPs.ExternalIPs != nil {
		if ipFamily == v1.IPv4Protocol {
			return service.IPs.ExternalIPs.GetV4(), nil
		} else if ipFamily == v1.IPv6Protocol {
			return service.IPs.ExternalIPs.GetV6(), nil
		}
	}
	return nil, errors.New(fmt.Sprint("ipfamily - not supported", ipFamily))
}
```
@jayunit100 something like this ?

Reply:
> wait can't this just be IPv4 or IPv6 then

Entries in an IPSet can only belong to one IPFamily, so I loop service creation logic for both the ipFamilies.
```
// iterate over ipFamily
for _, ipFamily := range []v1.IPFamily{v1.IPv4Protocol, v1.IPv6Protocol} {
    / /iterate over NodeIPs
    for _, nodeIP := range getNodeIPs(ipFamily) {
        ....
    }
}
```
And then I pick the relevant IPSet accordingly 
```
kubeNodePortTCPIPSet  = map[v1.IPFamily]string{
    v1.IPv4Protocol: "KUBE-NODE-PORT-TCP",
    v1.IPv6Protocol: "KUBE-6-NODE-PORT-TCP",
}
```